components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLModelHandler.java

```diff
@@ -355,6 +355,20 @@ public String streamingPredict(int tenantId, String userName, long modelId, Stri
             throw new MLModelHandlerException(msg);
         }
 
+        // Validate numerical feature type in predict dataset
+        for (Feature feature: builtModel.getFeatures()) {
+            if (feature.getType().equals(FeatureType.NUMERICAL)) {
+                int actualIndex = builtModel.getNewToOldIndicesList().indexOf(feature.getIndex());
+                for (String[] dataPoint: data) {
+                    if(!isNumeric(dataPoint[actualIndex])) {
+                        String msg = String.format("Invalid value: %s for the feature: %s",
```

Comment:
Yes, cause in the prediction dataset, we don't usually have the header, so it'll be helpful.